### ISSUE TITLE
Question about Ldiv code

### ISSUE THREAD

**Issue by darcula1993** [PRIS-CV/Mutual-Channel-Loss#10]
Brilliant idea!

I am trying to re-implement the code using tensorflow and I have some questions about Ldiv part:

`loss_2 = 1.0 - 1.0*torch.mean(torch.sum(branch,2))/cnum # set margin = 3.0`

1. what does the "margin" mean here?
2. Since you are using torch.mean, why do divide cnum again?
2. You are using 1 - Ldiv,which make the loss function become `Lmc = Ldis + lambda - lambda * Ldiv`
It seems not same as the paper description: `Lmc = Ldis - lambda * Ldiv`.

Thank you very much for any response!

**Comment by dongliangchang** [comment]
1 . Sorry for this, there  has a typo. you can ignore "set margin =3.0". In here, margin == cnum == the channels  belonging to each class.
2.  The upper-bound of Ldiv is equal to ξ , while the lower-bound is 1.   I use  /cnum to change the range from ( 1, ξ) to (1/ ξ), 1). Then ues 1 - Ldiv for optimize, It's just my code habit.   Loss = 1 - Ldiv  and Loss = -Ldiv, the only different is the hyper-parameter of it.
3. I think   Ldis - lambda * Ldiv  ==  Ldis + lambda * (1 -  Ldiv)  == Ldis + lambda - lambda * Ldiv, because lambda is noly a number, lambda.requires_grad == False.

**Comment by darcula1993** [comment]
I see.Thank you very much!
